app/controllers/ex_links_controller.rb

```diff
@@ -0,0 +1,96 @@
+class ExLinksController < ApplicationController
+  before_action :set_ex_link, only: [:show, :edit, :update, :destroy]
+
+  # GET /ex_links
+  def index
+    if params[:tag]
+      @ex_links = query_by_tags(params[:tag])
+    else
+      @ex_links = ExLink.all
+    end
+  end
+
+  # GET /ex_links/1
+  def show
+  end
+
+  # GET /ex_links/new
+  def new
+    @ex_link = ExLink.new
+  end
+
+  # GET /ex_links/1/edit
+  def edit
+  end
+
+  # POST /ex_links
+  def create
+    @ex_link = ExLink.new(ex_link_params)
+    if @ex_link.save
+      redirect_to @ex_link, notice: 'Ex link was successfully created.'
+    else
+      render :new
+    end
+  end
+
+  # PATCH/PUT /ex_links/1
+  def update
+    if @ex_link.update(ex_link_params)
+      redirect_to @ex_link, notice: 'Ex link was successfully updated.'
+    else
+      render :edit
+    end
+  end
+
+  # DELETE /ex_links/1
+  def destroy
+    @ex_link.destroy
+    redirect_to ex_links_url, notice: 'Ex link was successfully destroyed.'
+  end
+
+  # QUERY_BY_TAGS
+  # TODO1: Now returns even if tag is substring of other
+  # TODO1: e.g. Returns link which has tag 'bullsh' on 'bull' query
+  def query_by_tags(tags_string)
```

Comment:
This is a good example of something that should be done in the database. Right now you are loading every tag in the database for each request and processing them in unstructured form, which is suboptimal. Mysql has all this logic built-in, so you should prefer to use that.

Also, this is model logic since it deals with loading records from the database.

For a reasonable example, look at this code which does a full-text search in database space: https://github.com/fsek/web/blob/jforberg/coursescore/app/models/course.rb